projects/dynamicforms/src/util/validators/default-form-validators.ts

```diff
@@ -11,7 +11,10 @@ export class DefaultFormValidators extends Validators {
   static phoneNumberRegex = /^(?:\d{6,20})?$/;
 
   static required(control: AbstractControl) {
-    const valid = !!control.value?.trim();
+    const valid =
+      typeof control.value === 'object'
+        ? control.value.length > 0
```

Comment:
In the case of uploaded files `control.value` is an array of File objects, so I add additional check for this specific cases.

Reply:
The other solution would be `const valid = control.value || !!control.value?.trim();`, but in that case I need to refactor `removeAll` method from upload component and change `this.fileList = []` to `this.fileList = null`

Reply:
I would rather we make separate validation method instead of overriding default Angular `required` method. Granted, that would mean that in each form definition we would have to add this method where applicable, but in my opinion that would be better than overriding.
Also not sure if I get this condition (probably I'm missing something, but...):
If control value is of object type -> then control's length is more than 0? You don't set anything after checking the condition? Won't this always return false if control is of type object?

Reply:
If control value is object type -> than it checks condition `control.value.length > 0` and returns true or false to `valid` variable. Maybe we could discuss about this tomorrow.